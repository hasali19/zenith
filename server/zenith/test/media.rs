use insta::assert_json_snapshot;
use test_macros::test;

use crate::{with_app, TestApp};

macro_rules! test_snapshot {
    ($name:ident, $path:expr) => {
        #[test(with_app)]
        async fn $name(app: TestApp) {
            assert_json_snapshot!(app.get($path).await);
        }
    };
}

test_snapshot!(get_all_movies, "/movies");
test_snapshot!(get_movie_by_id, "/movies/1");
test_snapshot!(get_all_tv_shows, "/tv/shows");
test_snapshot!(get_show_by_id, "/tv/shows/4");
test_snapshot!(get_seasons_for_show, "/tv/shows/4/seasons");
test_snapshot!(get_season_by_id, "/tv/seasons/7");
test_snapshot!(get_episodes_for_show, "/tv/shows/4/episodes");
test_snapshot!(get_episodes_for_season, "/tv/seasons/7/episodes");
test_snapshot!(get_episode_by_id, "/tv/episodes/9");
