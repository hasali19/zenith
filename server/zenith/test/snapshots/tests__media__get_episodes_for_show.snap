---
source: server/zenith/test/media.rs
assertion_line: 21
expression: "app.get(\"/tv/shows/4/episodes\").await"
---
[
  {
    "air_date": null,
    "backdrop": null,
    "episode_number": 1,
    "external_ids": {
      "tmdb": null
    },
    "id": 9,
    "name": "Episode 1",
    "overview": null,
    "poster": null,
    "season_id": 7,
    "season_number": 1,
    "show_id": 4,
    "show_name": "Test Show 1",
    "thumbnail": null,
    "user_data": {
      "is_watched": false,
      "last_watched_at": null,
      "position": null
    },
    "video_info": {
      "duration": 0.0,
      "format": null,
      "path": "/path/to/Test Episode 1"
    }
  },
  {
    "air_date": null,
    "backdrop": null,
    "episode_number": 2,
    "external_ids": {
      "tmdb": null
    },
    "id": 10,
    "name": "Episode 2",
    "overview": null,
    "poster": null,
    "season_id": 7,
    "season_number": 1,
    "show_id": 4,
    "show_name": "Test Show 1",
    "thumbnail": null,
    "user_data": {
      "is_watched": false,
      "last_watched_at": null,
      "position": null
    },
    "video_info": {
      "duration": 0.0,
      "format": null,
      "path": "/path/to/Test Episode 2"
    }
  },
  {
    "air_date": null,
    "backdrop": null,
    "episode_number": 1,
    "external_ids": {
      "tmdb": null
    },
    "id": 11,
    "name": "Episode 1",
    "overview": null,
    "poster": null,
    "season_id": 8,
    "season_number": 2,
    "show_id": 4,
    "show_name": "Test Show 1",
    "thumbnail": null,
    "user_data": {
      "is_watched": false,
      "last_watched_at": null,
      "position": null
    },
    "video_info": {
      "duration": 0.0,
      "format": null,
      "path": "/path/to/Test Show 1/S02E01.mp4"
    }
  },
  {
    "air_date": null,
    "backdrop": null,
    "episode_number": 2,
    "external_ids": {
      "tmdb": null
    },
    "id": 12,
    "name": "Episode 2",
    "overview": null,
    "poster": null,
    "season_id": 8,
    "season_number": 2,
    "show_id": 4,
    "show_name": "Test Show 1",
    "thumbnail": null,
    "user_data": {
      "is_watched": false,
      "last_watched_at": null,
      "position": null
    },
    "video_info": {
      "duration": 0.0,
      "format": null,
      "path": "/path/to/Test Show 1/S02E02.mp4"
    }
  }
]
